cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -136,17 +136,22 @@ func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticP
 	}
 
 	// ensure etcd certs are generated for etcd and kube-apiserver
+	if component == constants.Etcd || component == constants.KubeAPIServer {
```

Comment:
Why CreateEtcdCACertAndKeyFiles needs to run when upgrading the api server?

Reply:
IIUC, `CreateEtcdServerCertAndKeyFiles` in L145 and `CreateAPIServerEtcdClientCertAndKeyFiles` in L153 depends on etcd CA to generate their cert and key files, `CreateEtcdCACertAndKeyFiles` ensures etcd CA files are existing or generated.

Reply:
Makes sense

/hold cancel

Reply:
Peter explained correctly what I was thinking :+1:

The happy path would work without it, but if you ever wanted the apiserver to start independently of etcd, it would fail.